### PR TITLE
NIAD-1153: Implemented fixes for encoding and environment variable check

### PR DESCRIPTION
**Pull request by mmackiewicz1**
Query params are properly encoded and environment variable is properly checked.

### CODE REVIEW

File: service/src/main/resources/application.yml [hunk 1]
```diff
@@ -10,6 +10,9 @@ logging:
         cloud:
           gateway: ERROR
 
+zuul:
```
Comment (mattd-kainos) [inline service/src/main/resources/application.yml]: Does this do anything? We're using spring cloud gateway and not zuul.

Comment (mmackiewicz1) [reply service/src/main/resources/application.yml]: Sorry, removed. Leftover of previous experiments.

File: service/src/main/java/uk/nhs/adaptors/gpc/consumer/utils/QueryParamsEncoder.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.util.UriComponentsBuilder;
+import org.springframework.web.util.UriUtils;
+
+public final class QueryParamsEncoder {
+    private QueryParamsEncoder() {
+    }
+
+    public static void encodeQueryParams(ServerWebExchange exchange) {
```
Comment (msolecka) [inline service/src/main/java/uk/nhs/adaptors/gpc/consumer/utils/QueryParamsEncoder.java:18]: Could you create tests for that class ?

File: service/src/intTest/java/uk/nhs/adaptors/gpc/consumer/CloudGatewayRouteBaseTest.java [hunk 3]
```diff
@@ -21,13 +21,19 @@
...
     protected static final String SSP_FROM_HEADER = "Ssp-From";
     protected static final String SSP_TO_HEADER = "Ssp-To";
     protected static final String SSP_INTERACTION_ID_HEADER = "Ssp-InteractionID";
     protected static final String SSP_TRACE_ID_HEADER = "Ssp-TraceID";
     protected static final String ANY_STRING = "any";
+    protected static final String GET_DOCUMENT_URI = "/GP0001/STU3/1/gpconnect/documents/Binary/07a6483f-732b-461e-86b6-edb665c45510";
+    protected static final String EXPECTED_DOCUMENT_BODY = "{\"resourceType\": \"Binary\","
+        + "\"id\": \"07a6483f-732b-461e-86b6-edb665c45510\","
+        + "\"contentType\": \"application/msword\","
+        + "\"content\": \"response content\"}";
```
Comment (andrewwa-kainos) [inline service/src/intTest/java/uk/nhs/adaptors/gpc/consumer/CloudGatewayRouteBaseTest.java:34]: Ode to [text blocks](https://openjdk.java.net/jeps/355), amirite?

**Review by andrewwa-kainos — APPROVED** [review]

File: service/src/main/java/uk/nhs/adaptors/gpc/consumer/utils/QueryParamsEncoder.java (hunk 2 above)
Comment (mmackiewicz1) [reply service/src/main/java/uk/nhs/adaptors/gpc/consumer/utils/QueryParamsEncoder.java:18]: Done

File: service/src/intTest/java/uk/nhs/adaptors/gpc/consumer/CloudGatewayRouteBaseTest.java (hunk 3 above)
Comment (mmackiewicz1) [reply service/src/intTest/java/uk/nhs/adaptors/gpc/consumer/CloudGatewayRouteBaseTest.java]: Added indents to jsons.

**Review by mattd-kainos — APPROVED** [review]

**Review by Broom94 — APPROVED** [review]

**Pull request merged**